analysis/src/Hint.ml

```diff
@@ -119,4 +119,60 @@ let inlay ~path ~pos ~maxLength ~debug =
                | Some value ->
                  if String.length label > value then None else Some result
                | None -> Some result)
-             | None -> None)))
\ No newline at end of file
+             | None -> None)))
+
+let codeLens ~path ~debug =
+  let lenses = ref [] in
+  let push loc =
+    let range = Utils.cmtLocToRange loc in
+    lenses := range :: !lenses
+  in
+  (* Code lenses are only emitted for functions right now. So look for value bindings that are functions,
+     and use the loc of the value binding itself so we can look up the full function type for our code lens. *)
+  let value_binding (iterator : Ast_iterator.iterator)
```

Comment:
This creates an AST iterator targeting _value bindings_, `let whatever = ...`. Since we're only looking for function definitions, we can target value bindings only, since that's the only place they'll appear.